### PR TITLE
changes to make gluon2019.1 smaller

### PR DESCRIPTION
**Pull request by skorpy2009**
https://gluon.readthedocs.io/en/v2019.1/releases/v2018.2.html#gluon-ebtables-limit-arp-enabled-by-default

### CODE REVIEW

File: site.conf
Comment (blocktrron) [inline site.conf:35]: While i think this is a sane choice, it does not impact the firmware size at all, as the code is still there, even with the password capability disabled.

File: site.mk
Comment (blocktrron) [inline site.mk:19]: Does not affect size, however selected as a dependency of `gluon-mesh-vpn-core` already.

File: site.mk
Comment (blocktrron) [inline site.mk]: While i think removing button-bind is a good idea, as the behavior is not consistent across platforms (hardware driven switch-leds, uncontrollable Power-LEDs), a cleanup of selected config states should be done (if this package has a persistent state).

File: site.mk
Comment (blocktrron) [inline site.mk]: This has the potential of breaking mesh-links on update. You probably need to write a migration package to remove persisting of channels on e.g. a pending scheduled domain-switch.

**Review by blocktrron — COMMENTED** [review]
Reviewing the parts which were obviously not auto-generated as i was requested for review. No authority on the ultimate decision here.

File: site.mk
Comment (skorpy2009) [reply site.mk:19]: So can still be drooped out of here... Even when it does not save space

File: site.conf
Comment (skorpy2009) [reply site.conf:35]: So will remove it for sanity reasons...

File: modules
Comment (margau) [inline modules:6]: Should be "discussed" on a wider level? Maybe reinclude when we have BATMAN 15 and the feature is upstream (https://github.com/freifunk-gluon/gluon/pull/1649)

File: site.conf
Comment (margau) [inline site.conf:32]: Should be included in Release Notes - and why.

File: site.mk
Comment (margau) [inline site.mk]: If feasible I think we should keep limit-arp, at least as long as we use Singledomain on a large scale.

File: site.mk
Comment (margau) [inline site.mk:10]: What does it do in current releases? Not sure if we don't break anything...

**Review by margau — CHANGES_REQUESTED** [review]
Regarding specific_site: What are the changes, on which information/feature base? Would love some infos regarding this.

File: modules
Comment (skorpy2009) [reply modules:6]: We should discuss this again when this feature is upstream

File: site.conf
Comment (skorpy2009) [reply site.conf:32]: Why: this is a security risk
Release Notes: YES!!!

File: site.mk
Comment (skorpy2009) [reply site.mk:10]: This is now included by default so not needed anymore here

**Comment by skorpy2009** [comment]
@margau https://git.darmstadt.ccc.de/ffda/site/blob/master/contrib/genpkglist.py this magic script is used for specific_site

File: modules
Comment (oszilloskop) [reply modules:6]: Please discussed this 'removing' with the big mesh maintainer. It looks like the package benefit is not understood.

File: site.mk
Comment (oszilloskop) [reply site.mk]: The pure "ffffm-button-bind" is a very helpfull package to protect node maintainer to search a proplem that doesn't really exist ([Freifunk housekeeper effect ](https://forum.freifunk.net/t/firmware-anderungswunsch-wifi-taste-deaktivieren/9009)).
"ffffm-button-bind" is one of the first simple base version of that kind of packages. Please do not compare it with the extended Gluon PR version.

Comment (oszilloskop) [reply site.mk]: "ffffm-banner" contains two tool scripts (nodeinfo / meshinfo) for the analysis of node and mesh topologies they should not be underestimated.

**Comment by oszilloskop** [comment]
Does this script make Gluon v2019 smaller?

File: site.mk
Comment (margau) [reply site.mk]: Regarding the tool scripts and button bind: I don't really see why we need our own tools for this, would prefer going into the migration with as little (custom) overhead as possible, and Upstream the tools/button bind if helpful (personally never used them).

File: site.conf
Comment (margau) [reply site.conf]: > @margau https://git.darmstadt.ccc.de/ffda/site/blob/master/contrib/genpkglist.py this magic script is used for specific_site

Would suggest including the script in our site/contrib for transparency. Currently it's not clear where does it come from.

**Comment by margau** [comment]
> @margau https://git.darmstadt.ccc.de/ffda/site/blob/master/contrib/genpkglist.py this magic script is used for specific_site

@skorpy2009 Would suggest including the script in our site/contrib for transparency. Currently it's not clear where does it come from.

File: site.mk
Comment (oszilloskop) [reply site.mk]: Regarding the **tool scripts**: There is no comparable script upstream now. The scripts are a collection of useful snippets from other communities and have been expanded based on own experience and needs as mesh maintainers.

Regarding **button-bind**: There is an upstream PR. This PR based on my/our Package. Currently the PR is more unwieldy and not very user friendly in my opinion.

**Comment by oszilloskop** [comment]
But then also with additions already established in our community. There were several requests from the local community for the integration of tools such as iperf3, tcpdump, joe, nano and more tools. These have been included in our image in Frankfurt for a long time (as long as the Flash has enough space).

**Comment by margau** [comment]
Would suggest to keep button bind until available upstream, and then merge the remaining parts. The dependency generation script should be added to contrib before @skorpy2009.

**Comment by margau** [comment]
Additionally: Would Merge into "legacy-batman-migration", as the migration is handled there.

**Comment by skorpy2009** [comment]
Now vs "legacy-batman-migration"

**Comment by oszilloskop** [comment]
Because there was no reaction to https://github.com/freifunk-ffm/site-ffffm/pull/42#discussion_r327480449, here again the objection I already mentioned there.

> Regarding the **tool scripts**: There is no comparable script upstream now. The scripts are a collection of useful snippets from other communities and have been expanded based on own experience and needs as mesh maintainers.

"ffffm-banner_legacy" contains two important tool scripts (nodeinfo / meshinfo) for the analysis of node and mesh topologies they should not be underestimated.

**Comment by margau** [comment]
> Because there was no reaction to [#42 (comment)](https://github.com/freifunk-ffm/site-ffffm/pull/42#discussion_r327480449), here again the objection I already mentioned there.
> 
> > Regarding the **tool scripts**: There is no comparable script upstream now. The scripts are a collection of useful snippets from other communities and have been expanded based on own experience and needs as mesh maintainers.
> 
> "ffffm-banner_legacy" contains two important tool scripts (nodeinfo / meshinfo) for the analysis of node and mesh topologies they should not be underestimated.

Don't really see the need - it's not in upstream gluon, and other "Freifunk-Firmware" perfectly works without it. 

If needed, it would suggest to add it to upstream gluon instead of maintaining "nodeinfo" and "meshinfo" on a local level.

**Comment by oszilloskop** [comment]
I really see the need - and our "Freifunk-Firmware" perfectly works with it.
As I say before the using of that tools should not be underestimated.

I don't have a problems to maintain this local package.

**Comment by skorpy2009** [comment]
This is not about kicking out packages for ever. This is about saving as much space as possible to be closer to the save side on the BATMAN migration

**Comment by oszilloskop** [comment]
You're contradicting yourself now. You say once only if it is upstream, and once only temporarily to have more flash space during the BATMAN migration.

But anyway, as you can see from the results of my investigations (https://github.com/freifunk-ffm/site-ffffm/issues/43), in the current situation the free space of the flash will not be of any use.

**Review by margau — APPROVED** [review]
From my side everything is fine now

**Comment by margau** [comment]
> You're contradicting yourself now. You say once only if it is upstream, and once only temporarily to have more flash space during the BATMAN migration.
> 
> But anyway, as you can see from the results of my investigations (#43), in the current situation the free space of the flash will not be of any use.

I see multiple aspects:
- We should minimize the overhead for the migration, as @blocktrron stated earlier
- We should stay as near as possible to upstream gluon, and not maintain "general tools" as our own overlay. This minimizes our Overhead and we have less "local stuff" that needs attention and possibly breaking things.

Considering both of those aspects, I'm approving this PR.

**Comment by christf** [comment]
On Tue, Oct 01, 2019 at 12:54:45AM -0700, Marvin Gaube wrote:
>> You're contradicting yourself now. You say once only if it is upstream, and once only temporarily to have more flash space during the BATMAN migration.

>> But anyway, as you can see from the results of my investigations (#43), in the current situation the free space of the flash will not be of any use.
>
>I see multiple aspects:
>- We should minimize the overhead for the migration, as @blocktrron stated earlier
It is not consensus that these packages actually create "overhead" 
whatever that means.

>- We should stay as near as possible to upstream gluon, and not maintain "general tools" as our own overlay. This minimizes our Overhead and we have less "local stuff" that needs attention and possibly breaking things.
This is not consensus and given the current state of the gluon project 
it is doubtful at best whether this course of action even makes sense.

Clearly visible, the direction for the migration and the roadmap after 
need to be *agreed* and written down.

I get it, you are the new guy and you want to get something done. 
Great! Why don't you start by asking a question about the motivation and 
the start taking the needs of those people seriously that are working 
their a** off in the field?

-- 
()  ascii ribbon campaign - against html e-mail
/\  www.asciiribbon.org   - against proprietary attachments

**Comment by margau** [comment]
I think we have to sort our priorities here:
We don't need to discuss our current biggest problems - everybody involved in FFFFM knows, that we have big problems with the backbone and our Single Domain. We have to migrate to BATMAN 15, as it's the last Gluon-Version with v14.

But the biggest problem: We have a very instable v14-Backbone, which is probably never going to be finally fixed. So, my objective with this branch is to prepare the basics for the switch to proceed as fast as possible without loosing quality. This is not stable, this is dev. When this is going near "stable", everything is usually clearly communicated on the Mailing list.

I'm supporting this PR, because I agree with it on a technical perspective. It takes FFFFM forward, and makes it better. We already have the "not really working"-reputation from some sides.

If we discuss for weeks before merging changes into a "Dev-Branch", we're just not moving forward.

**Pull request merged**
